lib/queue.js

```diff
@@ -705,6 +707,9 @@ Queue.prototype.add = function(name, data, opts) {
   opts = _.cloneDeep(opts || {});
   _.defaults(opts, this.defaultJobOptions);
 
+  if (this.limiter && this.limiter.key) {
+    opts.jobId = Date.now() + ":" + _.get(data, this.limiter.key);
```

Comment:
Instead of Date.now() a uuid should be used, otherwise we risk to loose jobs that are added within the same millisecond

Reply:
also, the user may have specified already a custom jobId, in that case we should base the new jobId on the one specified by the user instead of uuid.